adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -934,9 +934,10 @@ class ADAMContext private (@transient val sc: SparkContext) extends Serializable
     // load vcf metadata
     val (sd, samples, headers) = loadVcfMetadata(filePath)
 
-    val vcc = new VariantContextConverter(Some(sd))
+    val vcc = new VariantContextConverter()
+    val convFn = vcc.makeHtsjdkGenotypeConverter(headers)
```

Comment:
I generally agree that the interface is a crufty. My goal here was to clean up the guts of VariantContextConverter first. That being said, we might as well do it in one pass.

> convFn seems like it should be hidden from the API to me, unless there are caller-modifiable parts to it.

I agree.

> Why not pass headers (after modifications), htsjdk context, and stringency into separate Converter classes/objects?

WRT headers, a small (but significant) optimization is to only build the conversion function once. Thus the headers should be passed to the class constructor.

Otherwise, I agree.